File: src/main/java/fmweckserver/FmWeckServer.java
package fmweckserver;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.io.IOException;

public class FmWeckServer {

    private final Logger log = LogManager.getLogger(FmWeckServer.class);

    // TODO: implement proper shut-down

    public void startFmWeckServer(int PORT) {
        ProcessBuilder processBuilder = new ProcessBuilder(
                "fm-weck",
                "server",
                "--port",
                Integer.toString(PORT),
                "--listen",
                "localhost");

        processBuilder.redirectErrorStream(true);
        try {
            Process process = processBuilder.start();
            log.info("fm-weck server started on port " + PORT);
            // TODO: redirect server output to IDE output-log
        } catch (IOException e) {
            // TODO: proper error handling
            log.error("Failed to start fm-weck server", e);
            throw new RuntimeException(e);
        }
    }

}

File: src/main/kotlin/fmweckserver/FmWeckServer.kt
package fmweckserver

import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import java.io.IOException

class FmWeckServer {
    lateinit var process : Process

    private val log: Logger = LogManager.getLogger(FmWeckServer::class.java)

    // TODO: implement proper shut-down
    fun startFmWeckServer(port: Int) {
        val processBuilder = ProcessBuilder(
            "fm-weck",
            "server",
            "--port",
            port.toString(),
            "--listen",
            "localhost"
        )
        processBuilder.redirectErrorStream(true)
        try {
            process = processBuilder.start()
            log.info("fm-weck server started on port $port")
            // TODO: redirect server output to IDE output-log
        } catch (e: IOException) {
            log.error("Failed to start fm-weck server", e)
            TODO("proper error handling (startFmWeckServer)")
        }
    }
}